src/query/util/queryhttp/queryhttp.go

```diff
@@ -74,46 +63,19 @@ type RegisterOptions struct {
 	PathPrefix string
 	Handler    http.Handler
 	Methods    []string
+	Middleware []mux.MiddlewareFunc
 }
 
 // Register registers an endpoint.
-func (r *EndpointRegistry) Register(
-	opts RegisterOptions,
-	middlewareOpts ...logging.MiddlewareOption,
-) error {
-	// Wrap requests with response time logging as well as panic recovery.
-	var (
-		route             *mux.Route
-		metrics           = newRouteMetrics(r.instrumentOpts)
-		middlewareOptions []logging.MiddlewareOption
-	)
-	postRequestOption := logging.WithPostRequestMiddleware(
-		logging.PostRequestMiddleware(func(
-			r *http.Request,
-			meta logging.RequestMiddlewareMetadata,
-		) {
-			if !meta.WroteHeader {
-				return
-			}
-
-			p, err := route.GetPathTemplate()
-			if err != nil {
-				p = "unknown"
-			}
-
-			counter, timer := metrics.metric(p, meta.StatusCode)
-			counter.Inc(1)
-			timer.Record(meta.Duration)
-		}))
-	middlewareOptions = append(middlewareOptions, postRequestOption)
-	middlewareOptions = append(middlewareOptions, middlewareOpts...)
-
-	wrapped := func(n http.Handler) http.Handler {
-		return logging.WithResponseTimeAndPanicErrorLogging(n, r.instrumentOpts,
-			middlewareOptions...)
+func (r *EndpointRegistry) Register(opts RegisterOptions) error {
+	if len(opts.Middleware) == 0 {
+		opts.Middleware = middleware.Default(r.instrumentOpts)
+	}
+	handler := opts.Handler
+	for i := len(opts.Middleware) - 1; i >= 0; i-- {
```

Comment:
important: installing the middleware

Reply:
can we leave a comment why this ordering is important?